### PR TITLE
Resolve conflicts in testsuite/

### PR DESCRIPTION
**Pull request by mshinwell**
There turned out to be a lot of changes in generated Lambda code, backtraces, type error messages etc., which I've basically paid no attention to here in the majority of instances.  We can promote these once the compiler builds and we can run the testsuite properly.

There are however some proper changes mixed in here.

### CODE REVIEW

**Comment by mshinwell** [comment]
@ncik-roberts is going to review

**Comment by ncik-roberts** [comment]
Reviewing

File: testsuite/tests/backtrace/backtrace_effects_nested.ml [hunk 1]
```diff
@@ -1,6 +1,4 @@
 (* TEST
-   * skip
-
```
Comment (ncik-roberts) [inline testsuite/tests/backtrace/backtrace_effects_nested.ml:3]: Why delete this? That seems to contradict #191

Comment (ncik-roberts) [reply testsuite/tests/backtrace/backtrace_effects_nested.ml:3]: I'll resolve by restoring this and similar, and let you object before merging

File: testsuite/tests/shapes/comp_units.ml [hunk 2]
```diff
@@ -183,166 +182,6 @@ end
...
- "With_constraining_constraint"[module] ->
-   {<.16>
-    "M"[module] ->
-      {<.13>
-       "t"[type] ->
-         CU Stdlib . "Set"[module] . "Make"[module](
-         CU Stdlib . "Int"[module])<.13> . "t"[type];
-       };
-    };
->>>>>>> ocaml/5.1
```
Comment (ncik-roberts) [inline testsuite/tests/shapes/comp_units.ml:345]: Any reason we shouldn't just take `tip-5` here? (Do you know a reason why we wouldn't want to test this?)

File: testsuite/tests/shapes/functors.ml [hunk 3]
```diff
@@ -238,229 +237,6 @@ module Big_to_small2 : B2S = functor (X : Big) -> struct include X end
...
-|}]
-
-module Big_to_small2 : B2S = functor (X : Big) -> struct include X end
-[%%expect{|
-{
- "Big_to_small2"[module] ->
-   Abs<.42>(X/390, {
-                    "t"[type] -> X/390<.41> . "t"[type];
-                    });
->>>>>>> ocaml/5.1
```
Comment (ncik-roberts) [inline testsuite/tests/shapes/functors.ml:463]: Same question as above.

**Review by ncik-roberts — APPROVED** [review]
I pushed a number of corrections &mdash; please take a look at them and let me know if you disagree with any of them.

We can consider this approved except for the unresolved question of whether we need to restore `skip` on the effect tests.

File: testsuite/tests/backtrace/backtrace_effects_nested.ml (hunk 1 above)
Comment (mshinwell) [reply testsuite/tests/backtrace/backtrace_effects_nested.ml]: I think that was just a mistake, these should be skipped.

**Comment by mshinwell** [comment]
I think this is ok except for `931c20c148291c5de92d2357a8d9f46abdde104d ` which appears to have conflict markers on the + side of the diff?

**Comment by mshinwell** [comment]
LGTM

**Pull request merged**